test/mysql.test.js

```diff
@@ -235,6 +245,31 @@ describe('mysql', function() {
         });
       });
     });
+
+    it('isNewInstance should be undefined for after save hook', function(done) {
+      var student = {name: 'Joe', age: 20};
+      var newStudent = {};
+      Student.create(student, function(err, createdStudent) {
+        if (err) return done(err);
+        newStudent.id = createdStudent.id;
+        newStudent.name = 'Hannah';
+        newStudent.age = 25;
+        Student.replaceOrCreate(newStudent, function(err, s) {
+          if (err) return done(err);
+          s.id.should.equal(newStudent.id);
+          s.name.should.equal(newStudent.name);
+          s.age.should.equal(newStudent.age);
+        });
+
+        Student.observe('before save', function(ctx, next) {
```

Comment:
I am not sure how to make sure they are called in the same tick as `Student.replaceOrCreate()` here. Should I call them synchronously in the following order (using callbacks or async.series)?
`Student.observe('before save')` -> `Student.observe('after save') -> `Student.replaceOrCreate`